example/tests/test_nested_errors.py

```diff
@@ -0,0 +1,212 @@
+                    'comments': [
+                        {
+                        }
+                    ]
+                }
+            }
+        }
+
+        self.perform_error_test(data, '/data/attributes/comments/0/body')
+
+    def test_second_level_dict_error(self):
+        data = {
+            'data': {
+                'type': 'entries',
+                'attributes': {
+                    'blog': self.blog.pk,
+                    'body_text': 'body_text',
+                    'headline': 'headline',
+                    'comment': {}
+                }
+            }
+        }
+
+        self.perform_error_test(data, '/data/attributes/comment/body')
+
+    def test_third_level_array_error(self):
+        data = {
+            'data': {
+                'type': 'entries',
+                'attributes': {
+                    'blog': self.blog.pk,
+                    'body_text': 'body_text',
+                    'headline': 'headline',
+                    'comments': [
+                        {
+                            'body': 'test comment',
+                            'attachments': [
+                                {
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        }
+
+        self.perform_error_test(data, '/data/attributes/comments/0/attachments/0/data')
+
+    def test_third_level_custom_array_error(self):
+        data = {
+            'data': {
+                'type': 'entries',
+                'attributes': {
+                    'blog': self.blog.pk,
+                    'body_text': 'body_text',
+                    'headline': 'headline',
+                    'comments': [
+                        {
+                            'body': 'test comment',
+                            'attachments': [
+                                {
+                                    'data': 'text'
+                                }
+                            ]
+                        }
+                    ]
+                }
+            }
+        }
+
+        self.perform_error_test(data, '/data/attributes/comments/0/attachments/0/data')
+
+    def test_third_level_dict_error(self):
+        data = {
+            'data': {
+                'type': 'entries',
+                'attributes': {
+                    'blog': self.blog.pk,
+                    'body_text': 'body_text',
+                    'headline': 'headline',
+                    'comments': [
+                        {
+                            'body': 'test comment',
+                            'attachment': {}
+                        }
+                    ]
+                }
+            }
+        }
+
+        self.perform_error_test(data, '/data/attributes/comments/0/attachment/data')
+
+
+@pytest.mark.filterwarning('default::DeprecationWarning:rest_framework_json_api.serializers')
+def test_deprecation_warning(recwarn):
```

Comment:
in this module also a mix between unit test and pytest... why not just using pytest for the tests above as well?